src/OpenTelemetry.Abstractions/Trace/Event.cs

```diff
@@ -22,69 +22,45 @@ namespace OpenTelemetry.Trace
 
-    /// <inheritdoc/>
-    public sealed class Event : IEvent
+    /// <summary>
+    /// A text annotation associated with a collection of attributes.
+    /// </summary>
+    public sealed class Event
     {
         private static readonly ReadOnlyDictionary<string, object> EmptyAttributes =
                 new ReadOnlyDictionary<string, object>(new Dictionary<string, object>());
 
-        internal Event(string name, DateTimeOffset timestamp, IDictionary<string, object> attributes)
+        public Event(string name)
+            : this(name, PreciseTimestamp.GetUtcNow(), EmptyAttributes)
+        {
+        }
+
+        public Event(string name, DateTimeOffset timestamp)
+            : this(name, timestamp, EmptyAttributes)
+        {
+        }
+
+        public Event(string name, DateTimeOffset timestamp, IDictionary<string, object> attributes)
         {
             this.Name = name ?? throw new ArgumentNullException(nameof(name));
             this.Attributes = attributes ?? throw new ArgumentNullException(nameof(attributes));
             this.Timestamp = timestamp != default ? timestamp : PreciseTimestamp.GetUtcNow();
         }
 
-        /// <inheritdoc/>
-        public string Name { get; }
-
-        /// <inheritdoc/>
-        public DateTimeOffset Timestamp { get; }
-
-        /// <inheritdoc/>
-        public IDictionary<string, object> Attributes { get; }
-
         /// <summary>
-        /// Returns a new <see cref="Event"/> with the provided name.
+        /// Gets the <see cref="Event"/> name.
         /// </summary>
-        /// <param name="name">The text name for the <see cref="Event"/>.</param>
-        /// <returns>A new <see cref="Event"/> with the provided name.</returns>
-        /// <exception cref="ArgumentNullException">If <c>name</c> is <c>null</c>.</exception>
-        public static IEvent Create(string name)
-        {
-            return new Event(name, default, EmptyAttributes);
-        }
+        public string Name { get; }
 
         /// <summary>
-        /// Returns a new <see cref="Event"/> with the provided name.
+        /// Gets the <see cref="Event"/> timestamp.
         /// </summary>
-        /// <param name="name">The text name for the <see cref="Event"/>.</param>
-        /// <param name="timestamp">The timestamp of the <see cref="Event"/>.</param>
-        /// <returns>A new <see cref="Event"/> with the provided name.</returns>
-        /// <exception cref="ArgumentNullException">If <c>name</c> is <c>null</c>.</exception>
-        public static IEvent Create(string name, DateTimeOffset timestamp)
-        {
-            return new Event(name, timestamp, EmptyAttributes);
-        }
+        public DateTimeOffset Timestamp { get; }
 
         /// <summary>
-        /// Returns a new <see cref="Event"/> with the provided name and set of attributes.
+        /// Gets the <see cref="IDictionary{String, IAttributeValue}"/> collection of attributes associated with the event.
```

Comment:
Seems like the second generic argument mismatches in the doc and the type:
```suggestion
        /// Gets the <see cref="IDictionary{String, Object}"/> collection of attributes associated with the event.
```